definitions/npm/shelljs_v0.7.x/flow_>=v0.28.x/shelljs_v0.7.x.js

```diff
@@ -0,0 +1,381 @@
+  & ((pattern: $npm$shelljs$Pattern, subst: string, glob: string, ...rest: string[]) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$CmdSet =
+  & ((exitOnError: '-e' | '+e', rest: void) => void)
+  & ((verbose: '-v' | '+v', rest: void) => void)
+  & ((disableGlobbing: '-f' | '+f', rest: void) => void)
+  ;
+
+declare type $npm$shelljs$OptSort = $npm$shelljs$ICmdOptions<'-n' | '-r'>;
+declare type $npm$shelljs$CmdSort =
+  & ((options: $npm$shelljs$OptSort, glob: string, ...rest: string[]) => $npm$shelljs$String)
+  & ((glob: string, ...rest: string[]) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$CmdTail =
+  & ((num: number, glob: string, ...rest: string[]) => $npm$shelljs$String)
+  & ((glob: string, ...rest: string[]) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$CmdTempdir =
+  & ((rest: void) => string)
+  ;
+
+declare type $npm$shelljs$OptTest = '-b' | '-c' | '-d' | '-e' | '-f' | '-L' | '-p' | '-S';
+declare type $npm$shelljs$CmdTest =
+  & ((mode: $npm$shelljs$OptTest, path: string, rest: void) => boolean)
+  ;
+
+declare type $npm$shelljs$OptTouch = {
+  [key: '-a' | '-c' | '-m' ]: true;
+  '-d'?: string;
+  '-r'?: string;
+};
+declare type $npm$shelljs$CmdTouch =
+  & ((options: $npm$shelljs$OptTouch, glob: string, ...rest: string[]) => $npm$shelljs$String)
+  & ((glob: string, ...rest: string[]) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$CmdWhich =
+  & ((cmd: string, rest: void) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$PipeCat =
+  & ((rest: void) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$PipeExec =
+  & ((cmd: string, options: $npm$shelljs$OptExec & { async: true }, callback: $npm$shelljs$FunExec, rest: void) => $npm$shelljs$Async)
+  & ((cmd: string, options: $npm$shelljs$OptExec & { async: true }, rest: void) => $npm$shelljs$Async)
+  & ((cmd: string, options: $npm$shelljs$OptExecSync, rest: void) => $npm$shelljs$String)
+  & ((cmd: string, rest: void) => $npm$shelljs$String)
+  & ((cmd: string, callback: $npm$shelljs$FunExec, rest: void) => $npm$shelljs$Async)
+  ;
+
+declare type $npm$shelljs$PipeGrep =
+  & ((options: $npm$shelljs$OptGrep, pattern: $npm$shelljs$Pattern, rest: void) => $npm$shelljs$String)
+  & ((pattern: $npm$shelljs$Pattern, rest: void) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$PipeHead =
+  & ((num: number, rest: void) => $npm$shelljs$String)
+  & ((rest: void) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$PipeSed =
+  & ((pattern: $npm$shelljs$Pattern, subst: string, rest: void) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$PipeSort =
+  & ((options: $npm$shelljs$OptSort, rest: void) => $npm$shelljs$String)
+  & ((rest: void) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$PipeTail =
+  & ((num: number, rest: void) => $npm$shelljs$String)
+  & ((rest: void) => $npm$shelljs$String)
+  ;
+
+declare module 'shelljs' {
```

Comment:
> the main reason I have been using them in my libdefs so far is so that I can refer to types defined in libdefs for some module m0 in libdefs for another module m1.

Ah ok, yea that's definitely useful. If that's your purpose then fine by me until #16 is resolved